lib/telepoison.ex

```diff
@@ -35,14 +35,47 @@ defmodule Telepoison do
     :otel_propagator_text_map.inject(headers)
   end
 
+  @doc ~S"""
+  Performs a request using Telepoison with the provided `request` options.
+
+  See `HTTPoison.request/1` for further details regarding `request` options.
+
+  Will attempt to automatically set the `http.route` Open Telemetry metadata attribute by
+  removing the last part of the `request.url` (since this part typically contains dynamic data)
+  if the `resource_route` option is set to `:infer`.
+
+  If this behavior is not desirable, it can be set directly by using the aforementioned option.
+
+    ## Examples
+
+      iex> request = %HTTPoison.Request{
+      ...> method: :post,
+      ...> url: "https://www.example.com/users/edit/2",
+      ...> body: ~s({"foo": 3}),
+      ...> headers: [{"Accept", "application/json"}],
+      ...> options: [resource_route: :infer]}
+      iex> Telepoison.request(request)
+
+  """
   def request(%Request{options: opts} = request) do
     save_parent_ctx()
+
     span_name = Keyword.get_lazy(opts, :ot_span_name, fn -> compute_default_span_name(request) end)
 
+    resource_route =
+      case Keyword.get(opts, :resource_route) do
+        :infer ->
+          Telepoison.URI.infer_route_from_request(request)
+
+        route ->
+          route
+      end
+
     attributes =
       [
         {"http.method", request.method |> Atom.to_string() |> String.upcase()},
-        {"http.url", request.url}
+        {"http.url", request.url},
+        {"http.route", resource_route}
```

Comment:
This will set `{"http.route", nil}` if not provided. Not sure what OpenTelemetry then does with that, does it automatically drop attributes with `nil` values? In any case, maybe we should not set it at all when `resource_route` is not set.

Reply:
@LeartS certainly - updated the attribute name and how the `http.route` attribute is added in; not sure if it's a bit too ugly however. What do you think?